### DISCUSSION TITLE
Is possible to develop an extension with a hotkey?

### DISCUSSION THREAD

**Discussion by aurexav** [zed-industries/zed#17557] [Feature Requests]
I'm using zed_extension_api to develop a Zed extension.

1. It providea a hotkey,
1. Its main feature is select the content between the same characters, like `()`, `[]`, `{}`, `""`, `''`, `<>`, etc.
1. Press once, select the content between the same characters, press twice, select the content and the characters.
1. If conitnue to press, find the next level "bracket" and repeat the above operation.

Is this possible? I didn't find any useful tutorial. What's the enter point? How to control the editor's selection? How to bind this action with a hotkey?

It's really easy to implement for VSCode. But I don't know if this is possible in Zed.

**Comment by SomeoneToIgnore** [comment]
It's not to my knowledge, the extension API is very limited still.

**Comment by aurexav** [comment]
Move to https://github.com/zed-industries/zed/issues/17562.
